release-build.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+function showUsage
+{
+  echo -e "\nThis script is used to build a release"
+  echo
+  echo -e "Usage: ./`basename $0 $1 $2` actual_version next_version previous_version"
+  echo
+  echo -e "Example to build release 4.4.0 with next version 4.4.1 and previous version 4.2.5:"
+  echo -e "\t./`basename $0 $1` 4.4.0 4.4.1 4.2.5"
+  echo
+}
+
+if [ "$1" = "-h" ]
+then
+	showUsage
+	exit
+fi
+
+if [ $# -ne 3 ]
+then
+  showUsage
+  exit
+fi
+
+frombranch=origin/main
+versionbranch=main
+version=$0
+minorversion=0
+newversion=$version-$minorversion
+currentversion=$0-SNAPSHOT
+previousversion=$2
+nextversion=$1-SNAPSHOT
+
+
+git clone --recursive https://github.com/geonetwork/core-geonetwork.git \
+          geonetwork-$versionbranch
+cd geonetwork-$versionbranch
+
+
+git checkout -b $versionbranch $frombranch
+
+
+# Update version number (in pom.xml, installer config and SQL)
+./update-version.sh $currentversion $newversion
+# TODO: Check the JRE URL replacement
+
+# Generate list of changes
+cat <<EOF > docs/changes$newversion.txt
+================================================================================
+===
+=== GeoNetwork $version: List of changes
+===
+================================================================================
+EOF
+git log --pretty='format:- %s' $previousversion... >> docs/changes$newversion.txt
+
+# Then commit the new version
+git add .
+git commit -m "Update version to $newversion"
+git tag -a $version -m "Tag for $version release"
+
+# Build the new release
+mvn clean install -DskipTests -Pwar -Pwro4j-prebuild-cache
+
+cd datastorage
+mvn clean install -Drelease
+cd ..
+
+# Download Jetty and create the installer
+cd release
+mvn clean install -Djetty-download
+ant
+cd ..
+
+# Set version number to SNAPSHOT
+./update-version.sh $newversion $nextversion
+
+git add .
+git commit -m "Update version to $nextversion"
+
+
+if [[ ${OSTYPE:0:6} == 'darwin' ]]; then
+  md5 -r web/target/geonetwork.war > web/target/geonetwork.war.md5
+  md5 -r release/target/GeoNetwork-$newversion/geonetwork-bundle-$newversion.zip > release/target/GeoNetwork-$newversion/geonetwork-bundle-$newversion.zip.md5
+else
+  cd web/target && md5sum geonetwork.war > geonetwork.war.md5 && cd ../..
```

Comment:
Instead of `cd ../..`
You could use `cd -`
That way if ever the initial cd changes then `cd -` will still work.

Another alternative is to run it in a subshell so that it does not affect the current shell pwd
`( cd web/target && md5sum geonetwork.war > geonetwork.war.md5 )`

Reply:
Updated @ianwallen. Thanks